connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -61,24 +61,28 @@
                     " This controls the format of the values in messages written to or read from Kafka, and since this is" +
                     " independent of connectors it allows any connector to work with any serialization format." +
                     " Examples of common formats include JSON and Avro.";
-
+    @Deprecated
```

Comment:
These doc changes aren't sufficient due to the way Kafka configs work. Without a default provided, parsing the configs will throw an exception (before you ever even hit the code you have in `Worker.java`). You'd need to change the default value to actually be `JsonConverter`. (This is actually a bit limiting once you get to the parsed version of the config because you can't tell if the user actually *specified `JsonConverter` explicitly or is using the default; however, I think this is a limit of the `CLASS` type in `ConfigDef`. If it was just a `STRING` we could leave it empty by default, detect that and fall back to `JsonConverter`, and know based on this whether we should be using the user's settings for each converter or just specify our set of defaults.)

Reply:
Yes, I can understand that when we don't provide any defaults to the configs while defining the `ConfigDef`, it is considered as required and one has to configure that config explicitly. I believe by configuring the default as `JsonConverter`, we can still check whether some converter other than `JsonConverter` is provided by user. Although it will certainly be difficult to distinguish where this `JsonConverter` is set (its getting populated as default or provided by user), still the code will proceed further to follow the KIP and to use the user provided converters. I think it is the expected behavior as per KIP and setting the default for these deprecated configs should not break the existing users's code.